packages/common/src/utils/formatBalance.ts

```
@@ -16,4 +16,65 @@ export const formatBalance = (balance: bigint, decimals: number): string => {
   
     return fractionalStr ? `${integerPart}.${fractionalStr}` : integerPart.toString()
 }
-  
\ No newline at end of file
+  
+
+
+/**
+ * Converts a token amount from human-readable format to the smallest unit (e.g., DOT to Planck)
+ * @param value - The amount in human-readable format as a string (e.g., "10.5")
+ * @param decimals - The number of decimal places for the token (e.g., 10 for DOT, 12 for WND)
+ * @returns The amount in smallest unit as bigint
+ * @throws Error if the input is invalid or negative
+ * 
+ * @example
+ * ```typescript
+ * // Convert 10 DOT to Planck (10 * 10^10)
+ * const planck = parseUnits("10", 10); // Returns 100000000000n
+ * 
+ * // Convert 10.23234 DOT to Planck
+ * const planck = parseUnits("10.23234", 10); // Returns 102323400000n
+ * ```
+ */
+export function parseUnits(value: string, decimals: number): bigint {
+
+    if (typeof value !== 'string') {
+      throw new Error('Value must be a string');
+    }
+    
+
+    if (!Number.isInteger(decimals) || decimals < 0) {
+      throw new Error('Decimals must be a non-negative integer');
+    }
+  
+
+    if (!value.trim()) {
```

Can we please do all the checks above and throw error `Invalid value as an input`. This will be cleaner